modules/core/shared/src/main/scala/io/circe/Printer.scala

```diff
@@ -209,16 +207,19 @@ final case class Printer(
       else Printer.NoSizePredictor
 
     val writer = new Printer.AppendableByteBuffer(cs, predictor)
-    val folder = new AppendableByteBufferFolder(writer)
-
-    json.foldWith(folder)
+    printToAppendable(json, writer)
 
     writer.toByteBuffer
   }
 
   final def printToByteBuffer(json: Json): ByteBuffer =
     printToByteBuffer(json, StandardCharsets.UTF_8)
 
+  final def printToAppendable(json: Json, out: Appendable): Unit = {
+    val folder = new AppendableFolder(out)
+    json.foldWith(folder)
+  }
+
```

Comment:
Thanks for the PR!  I think circe has really tried to avoid public methods with side effecting signatures, so I'm fairly hesitant to introduce one. That being said,  I think if we mark the boundaries of safeness and let others know that thar be dragons.. then we can probably let it in.  So something like this:
```suggestion
  object Unsafe {
    final def printToAppendable(json: Json, out: Appendable): Unit = {
      val folder = new AppendableFolder(out)
      json.foldWith(folder)
    }
  }
```

And we should add some scaladoc to the method as well.  For instance, it should be noted that this method returns only after all JSON is written to the appendable and that any use of the `Appendable` outside this method voids the warranty 😆 .